api/src/sophia/model.rs

```diff
@@ -0,0 +1,212 @@
+//! Provide implementation of Sophia traits for Rio types.
+use crate::model::*;
+use sophia_api::ns::xsd;
+#[cfg(feature = "generalized")]
+use sophia_api::quad::Quad as SophiaQuad;
+use sophia_api::term::{RawValue, SimpleIri, TTerm, TermKind};
+
+impl<'a> TTerm for NamedNode<'a> {
+    #[inline]
+    fn kind(&self) -> TermKind {
+        TermKind::Iri
+    }
+    #[inline]
+    fn value_raw(&self) -> RawValue<'_> {
+        self.iri.into()
+    }
+    #[inline]
+    fn as_dyn(&self) -> &dyn TTerm {
+        self
+    }
+}
+
+impl<'a> From<&NamedNode<'a>> for SimpleIri<'a> {
+    #[inline]
+    fn from(other: &NamedNode<'a>) -> SimpleIri<'a> {
+        SimpleIri::new_unchecked(other.iri, None)
+    }
+}
+
+impl<'a> TTerm for BlankNode<'a> {
+    #[inline]
+    fn kind(&self) -> TermKind {
+        TermKind::BlankNode
+    }
+    #[inline]
+    fn value_raw(&self) -> RawValue<'_> {
+        self.id.into()
+    }
+    #[inline]
+    fn as_dyn(&self) -> &dyn TTerm {
+        self
+    }
+}
+
+impl<'a> TTerm for Literal<'a> {
+    #[inline]
+    fn kind(&self) -> TermKind {
+        TermKind::Literal
+    }
+    #[inline]
+    fn value_raw(&self) -> RawValue<'_> {
+        match self {
+            Literal::Simple { value } => *value,
+            Literal::LanguageTaggedString { value, .. } => *value,
+            Literal::Typed { value, .. } => *value,
+        }
+        .into()
+    }
+    #[inline]
+    fn datatype(&self) -> Option<SimpleIri<'_>> {
```

Comment:
Because I'm distracted, and that's what it should do! Thanks for spotting that...

Reply:
>  Why integrating all the Sophia adapters in Rio instead of having them in Sophia?

In the end, the `sophia` crate is meant to become an empty shell re-exporting stuff from other, more specialized crates. And while this is useful for quick prototyping, I would encourage developers of bigger projects to directly pick and chose the specialized package they really need.

Regarding trait impl vs. wrappers: my goal was not (only) to work around Rust limitations for implementing traits on foreign types. It was more generally to add *a layer of sophia-compliance atop Rio*. Rio provides parsers, it then makes sense that it implements, somehow, the `TripleParser`/`QuadParser` traits of Sophia. Since Rio and Sophia have slightly different notions of what a parser is, this needs a few wrappers, but these wrappers are slim enough (I think).

Of course, if you disagree, I will just leave the trait impl and move the wrappers to a separate crate in the Sophia repo.